lib/Cake/TestSuite/CakeTestCase.php

```diff
@@ -716,6 +716,78 @@ protected function skipUnless($condition, $message = '') {
 	// @codingStandardsIgnoreEnd
 
 /**
+ * Returns a mock object for the specified class.
+ *
+ * @param string $originalClassName The class name of the object to be mocked.
+ * @param array $methods By default, all methods of the given class are replaced
+ *   with a test double that just returns NULL unless a return value is configured
+ *   using will($this->returnValue()), for instance.
+ *   When the second (optional) parameter is provided, only the methods whose names
+ *   are in the array are replaced with a configurable test double. The behavior
+ *   of the other methods is not changed. Providing NULL as the parameter means
+ *   that no methods will be replaced.
+ * @param array $arguments The third (optional) parameter may hold a parameter
+ *   array that is passed to the original class' constructor (which is not replaced
+ *   with a dummy implementation by default).
+ * @param string $mockClassName The fourth (optional) parameter can be used to
+ *   specify a class name for the generated test double class.
+ * @param bool $callOriginalConstructor The fifth (optional) parameter can be
+ *   used to disable the call to the original class' constructor.
+ * @param bool $callOriginalClone The sixth (optional) parameter can be used
+ *   to disable the call to the original class' clone constructor.
+ * @param bool $callAutoload The seventh (optional) parameter can be used to
+ *   disable __autoload() during the generation of the test double class.
+ * @param bool $cloneArguments Not supported.
+ * @param bool $callOriginalMethods Not supported.
+ * @param string $proxyTarget Not supported.
+ * @return object
+ * @throws InvalidArgumentException When not supported parameters are set.
+ * @deprecated Use `getMockBuilder()` or `createMock()` in new unit tests.
+ * @see https://phpunit.de/manual/current/en/test-doubles.html
+ */
+	public function getMock($originalClassName, $methods = array(),
+			array $arguments = array(), $mockClassName = '',
+			$callOriginalConstructor = true, $callOriginalClone = true,
+			$callAutoload = true, $cloneArguments = false,
+			$callOriginalMethods = false, $proxyTarget = null) {
+		$phpUnitVersion = PHPUnit_Runner_Version::id();
+		if ($phpUnitVersion < '5.7.0') {
+			return parent::getMock($originalClassName, $methods, $arguments,
+					$mockClassName, $callOriginalConstructor, $callOriginalClone,
+					$callAutoload, $cloneArguments, $callOriginalMethods, $proxyTarget);
+		}
+		$MockBuilder = $this->getMockBuilder($originalClassName);
```

Comment:
I disagree with this statement. Just removing or omitting white lines which would visually group actual "parts" of the method does not remove any code smell, if there was one.

Sometimes a method does have parts, most often when the code cannot be extracted further into separate methods simply by the nature of the code.

And in case of ``getMock()`` here, we do have at least two of such parts:

1. part: Branching logic with check for version and return early.
2. part: The main mock building
3. part (potential): checking for invalid arguments and throwing exceptions, this could be done early, though I understand the current order.
4. part (arguable): the return statement (in CakePHP 3 we adpted a blank line before almost all return statements. While I personally do not agree with that in all those cases, it lead to mostly more readable code IMHO).

Disclaimer: I'm the co-developer or [PHPMD](https://phpmd.org/), that's why I care so much about messes in code and code smells.